pkg/apis/kops/validation/legacy.go

```diff
@@ -291,6 +291,31 @@ func ValidateCluster(c *kops.Cluster, strict bool) field.ErrorList {
 					}
 				}
 			}
+
+			// @ check that NodeLocalDNS addon is configured correctly
+			if c.Spec.KubeDNS.NodeLocalDNS != nil && c.Spec.KubeDNS.NodeLocalDNS.Enabled {
+				if c.Spec.KubeDNS.Provider != "CoreDNS" {
```

Comment:
@olemarkus I think this would be enough right? does this mean that the user must set explicitly the `Provider` field in the spec to `CoreDNS`?

Reply:
This should work, yes.